### PR TITLE
Added status_string method to return simple status strings

### PR DESCRIPTION
**Pull request by tamenol**
status_string returns the current status of the bus in a single word.

Ok, so I have no idea this is warrented to get merged. I wrote this function as I needed a single word for the status of the bus (just like you have in PCAN-View). And I didn't want to import .basic in my project to get the needed values, hence why I put it in the `PcanBus` class

### CODE REVIEW

File: can/interfaces/pcan/pcan.py [hunk 1]
```diff
@@ -528,6 +528,32 @@ def _detect_available_configs():
...
+        elif status == PCAN_ERROR_BUSHEAVY:
+            return "BUSHEAVY"
+        elif status == PCAN_ERROR_BUSWARNING:
+            return "BUSWARNING"
+        elif status == PCAN_ERROR_BUSPASSIVE:
+            return "BUSPASSIVE"
+        elif status == PCAN_ERROR_BUSOFF:
+            return "BUSOFF"
+        else:
+            return "NonImplementedError"
```
Comment (felixdivo) [inline can/interfaces/pcan/pcan.py]: I' return `None` in this case, to make it more explicit that something went wrong.

File: can/interfaces/pcan/pcan.py [hunk 2]
```diff
@@ -528,6 +528,32 @@ def _detect_available_configs():
             )
         return channels
 
+    def status_string(self):
```
Comment (felixdivo) [inline can/interfaces/pcan/pcan.py]: I'd replace the `rtype` comment with an actual type hint: 

```python
def status_string(self) -> Optional[str]:
```

**Comment by felixdivo** [comment]
Sorry for the long delay, do you still have tome to fix this?

**Comment by tamenol** [comment]
@felixdivo No problem. And yes, I'll have a look this weekend/in the course of next week.

File: can/interfaces/pcan/pcan.py (hunk 1 above)
Comment (tamenol) [reply can/interfaces/pcan/pcan.py]: Yes, good idea. However, when looking back on this, wouldn't it be better if I made a dictonary with all the errors and their strings in `basic.py` and retrieve them with this function, so the class in `pcan.py` doesn't get cluttered much?

Comment (tamenol) [reply can/interfaces/pcan/pcan.py]: @felixdivo see commit https://github.com/hardbyte/python-can/pull/725/commits/890128f40da623299f153d7e0fed1de028f2d417

**Comment by felixdivo** [comment]
Looks good to me.

**Comment by felixdivo** [comment]
Except [the formatter](https://black.readthedocs.io/en/stable/) is complaining [here](https://travis-ci.org/hardbyte/python-can/jobs/645876354#L321). You can fix it by executing `black .` in `python-can/can/`.

**Comment by felixdivo** [comment]
What a massive change 👍

**Pull request merged**